src/Backups/BackupCoordinationStageSync.cpp

```diff
@@ -115,7 +115,7 @@ BackupCoordinationStageSync::BackupCoordinationStageSync(
     , failure_after_host_disconnected_for_seconds(with_retries.getKeeperSettings().failure_after_host_disconnected_for_seconds)
     , finish_timeout_after_error(with_retries.getKeeperSettings().finish_timeout_after_error)
     , sync_period_ms(with_retries.getKeeperSettings().sync_period_ms)
-    , max_attempts_after_bad_version(with_retries.getKeeperSettings().max_attempts_after_bad_version)
+    , max_attempts_after_bad_version(with_retries.getKeeperSettings().max_attempts_after_bad_version + all_hosts.size())
```

Comment:
Please add  a comment that we need to add `all_hosts.size()` here because it's a valid case when other hosts create some Keeper nodes concurrently.